src/main.rs

```diff
@@ -33,10 +33,14 @@ struct Backend {
 impl LanguageServer for Backend {
     async fn initialize(&self, params: InitializeParams) -> Result<InitializeResult> {
         let mut world = self.world.write().await;
-        *world = Some(SystemWorld::new(
-            params.root_uri.unwrap().to_file_path().unwrap(),
-        ));
-
+        // Check if a folder is opened, if yes, use it as the root path
+        let root_path = if let Some(root) = params.root_uri {
+            root.to_file_path().unwrap()
+        } else {
+            PathBuf::new()
+        };
```

Comment:
Personal preference but I would rewrite this as a match statement:

```rust
let root_path = match params.root_uri {
    Some(root) => root.to_file_path().unwrap(),
    None => PathBuf::new()
};
```

or even to:

```rust
let root_path = params.root_uri.map(|root| root.to_file_path().unwrap()).unwrap_or_else(|| PathBuf::new());
```

I think either of those look more idiomatic.

Reply:
Yes definitly, I like your first choice. I am quiet new to Rust so thanks for the comment :)